### DISCUSSION TITLE
Are location and server directives automatically applied in the Advanced tab?

### DISCUSSION THREAD

**Discussion by looeee** [NginxProxyManager/nginx-proxy-manager#2183] [Q&A]
I've run into the `upstream sent too big header while reading response header from upstream` error with keycloak so I need to increase the `proxy_buffer_size` size. 

What I have seen is people simply pasting the following into the advanced tab for the proxy host. 

```
proxy_buffer_size 128k;
```

If I do this, where is it applied? Just to this host? Or all hosts?

Also, it seems like I only need this at `/auth/realms/master/protocol/openid-connect/auth` so can I use a location like this?

```
location /auth/realms/master/protocol/openid-connect/auth/ {
    proxy_buffer_size 16k;
}
```

**Comment by looeee** [comment]
OK so I figured this out. 
For anyone else coming here, the answer is no. You need to add a custom location and set it to `/` then apply the rules there.

**Comment by ghost** [comment]
Could you be more specific with your instructions please?Sent from my Galaxy
-------- Original message --------From: Lewy Blue ***@***.***> Date: 09/09/2022  04:18  (GMT+00:00) To: NginxProxyManager/nginx-proxy-manager ***@***.***> Cc: Subscribed ***@***.***> Subject: Re: [NginxProxyManager/nginx-proxy-manager] Are location and server directives automatically applied in the Advanced tab? (Discussion #2183) 
OK so I figured this out.
For anyone else coming here, the answer is no. You need to add a custom location and set it to / then apply the rules there.
